### PR TITLE
Fix telemetry and define artifacts folder

### PR DESCRIPTION
**Pull request by chcosta**
As @alexperovich  pointed out recently, telemetry was broken for Arcade. 

- Flow telemetry values through to telemetry.yml
- Fix publish artifact drop name

Examples of problems reported
https://devdiv.visualstudio.com/DevDiv/Default/_build/results?buildId=1763259
https://mc.dot.net/#/user/dotnet-mc-bot-2/$(helixsource)/$(helixtype)/20180611.2/workItem/Orchestration/analysis/external/Link

### CODE REVIEW

File: eng/build.yml [hunk 1]
```diff
@@ -10,34 +10,38 @@
...
     buildReason: ${{ parameters.buildReason }}
+    buildConfig: $(_BuildConfig)
     phaseName: ${{ parameters.agentOs }}
     dockerImage: ${{ parameters.dockerImage }}
     enableTelemetry: ${{ parameters.enableTelemetry }}
+    helixSource: official/dotnet/arcade/$(Build.SourceBranch)
+    ${{ if notIn(parameters.buildReason, 'IndividualCI', 'BatchedCI', 'PullRequest') }}:
+      helixType: build/product          
+    ${{ if in(parameters.buildReason, 'IndividualCI', 'BatchedCI', 'PullRequest') }}:
+      helixType: pr/dotnet/arcade/$(Build.SourceBranch)
```
Comment (alexperovich) [inline eng/build.yml]: This is not quite correct. helixType should always be "build/product". helixSource should be "official/{org}/{repo}/{branch}" for official runs and "pr/{org}/{repo}/{branch}" for pull request runs.

File: eng/common/templates/steps/telemetry.yml [hunk 2]
```diff
@@ -20,20 +23,21 @@ steps:
...
-          Configuration: $(_BuildConfig)
+          OperatingSystem: ${{ parameters.agentOs }}
+          Configuration: ${{ parameters.buildConfig }}
         condition: always()
       - bash: |
           /bin/bash ./telemetry/build/start.sh --build-uri "$BuildUri"
         workingDirectory: $(Build.SourcesDirectory)/eng/common
         displayName: Send Build Start Telemetry
         env:
           BuildUri: https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=$(Build.BuildId)&_a=summary
```
Comment (alexperovich) [inline eng/common/templates/steps/telemetry.yml]: We will need to change this to point to the VSTS account that is running the builds. I don't know how to get that from the environment in a build.

File: eng/build.yml (hunk 1 above)
Comment (chcosta) [reply eng/build.yml]: Thanks, my bad

File: eng/common/templates/steps/telemetry.yml (hunk 2 above)
Comment (chcosta) [reply eng/common/templates/steps/telemetry.yml]: I'll hunt around

File: eng/build.yml [hunk 3]
```diff
@@ -10,34 +10,38 @@
...
     buildReason: ${{ parameters.buildReason }}
+    buildConfig: $(_BuildConfig)
     phaseName: ${{ parameters.agentOs }}
     dockerImage: ${{ parameters.dockerImage }}
     enableTelemetry: ${{ parameters.enableTelemetry }}
+    helixType: build/product
+    ${{ if notIn(parameters.buildReason, 'IndividualCI', 'BatchedCI', 'PullRequest') }}:
+      helixSource: official/dotnet/arcade/$(Build.SourceBranch)
+    ${{ if in(parameters.buildReason, 'IndividualCI', 'BatchedCI', 'PullRequest') }}:
+      helixType: pr/dotnet/arcade/$(Build.SourceBranch)
```
Comment (alexperovich) [inline eng/build.yml]: This still says helixType

Comment (chcosta) [reply eng/build.yml]: yep, fixed that and additional extra var.  It doesn't look like the VSTS gate is firing for this PR anymore, I'm investigating why...

File: eng/common/templates/steps/telemetry.yml [hunk 4]
```diff
@@ -20,20 +23,21 @@ steps:
...
+          OperatingSystem: ${{ parameters.agentOs }}
+          Configuration: ${{ parameters.buildConfig }}
         condition: always()
       - bash: |
           /bin/bash ./telemetry/build/start.sh --build-uri "$BuildUri"
         workingDirectory: $(Build.SourcesDirectory)/eng/common
         displayName: Send Build Start Telemetry
         env:
-          BuildUri: https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=$(Build.BuildId)&_a=summary
+          BuildUri: $(System.TaskDefinitionsUri)$(System.TeamProject)/_build/index?buildId=$(Build.BuildId)&_a=summary
```
Comment (alexperovich) [inline eng/common/templates/steps/telemetry.yml]: Can we send an ask to VSTS to document this variable?

File: eng/common/templates/steps/telemetry.yml [hunk 5]
```diff
@@ -1,6 +1,9 @@
 
 parameters:
   agentOs: Windows_NT
+  helixSource: ''
```
Comment (jonfortescue) [inline eng/common/templates/steps/telemetry.yml]: should these have default values that are more obvious for logging?

File: eng/build.yml [hunk 6]
```diff
@@ -100,8 +103,12 @@ phases:
...
           inputs:
             PathtoPublish: '$(Build.StagingDirectory)/BuildLogs'
-            ArtifactName: $(_agentOs)_$(Agent.JobName)
             PublishLocation: Container
+            # Should be able to use a single definition of ArtifactName here instead of conditioned on 
+            # agentOs once variables are available in the template evaluation context.  
+            ${{ if eq(parameters.agentOs, 'Windows_NT') }}:
+              ArtifactName: Windows_NT_$(Agent.JobName)
+            ${{ if ne(parameters.agentOs, 'Windows_NT') }}:
+              ArtifactName: Linux_$(Agent.JobName)
```
Comment (jonfortescue) [inline eng/build.yml]: So then I'm guessing this isn't designed to work on OSX?

**Comment by chcosta** [comment]
Closing / re-opening to try to retrigger PR gates.

File: eng/common/templates/steps/telemetry.yml [hunk 7]
```diff
@@ -83,6 +88,7 @@ steps:
         displayName: Send Build Start Telemetry
         env:
           BuildUri: https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=$(Build.BuildId)&_a=summary
```
Comment (jonfortescue) [inline eng/common/templates/steps/telemetry.yml]: do we not need to modify the build URI like you did on line 39? Parameterizing it seems smart.

**Comment by chcosta** [comment]
VSTS gate failure appears to be fixed now.  I had to disable the triggers and then re-enable them to get them to fire again.  This has happened in the past.  I'll follow up with VSTS to ensure a fix is pending.

File: eng/build.yml (hunk 6 above)
Comment (chcosta) [reply eng/build.yml]: Correct, there's no OSX build leg in Arcade today.

File: eng/common/templates/steps/telemetry.yml (hunk 5 above)
Comment (chcosta) [reply eng/common/templates/steps/telemetry.yml]: I set defaults for these in `base.yml` as `unspecified`.  Seems reasonable to add defaults here as well.

Comment (chcosta) [reply eng/common/templates/steps/telemetry.yml]: Now, I'm remembering that my thinking was that if someone was using `base.yml` and didn't provide one of these values then I wanted to know that vs if someone directly used `telemetry.yml`, then they seem more of a "power-user" and I was ok letting themselves do weird things.  


Perhaps we could set the values to something like `undefined_defaulted_in_base.yml` and `undefined_defaulted_in_telemetry.yml`?  That might be overthinking it and simply setting `unspecified` is sufficient.  Thoughts?

File: eng/common/templates/steps/telemetry.yml (hunk 7 above)
Comment (chcosta) [reply eng/common/templates/steps/telemetry.yml]: For sure

File: eng/common/templates/steps/telemetry.yml (hunk 5 above)
Comment (jonfortescue) [reply eng/common/templates/steps/telemetry.yml]: I like the idea of differentiating between people using base vs. directly from telemetry. I think that using the two different defaults makes sense to me &ndash; more telemetry info is useful I think.

**Comment by jonfortescue** [comment]
@dotnet-bot test this please

**Comment by chcosta** [comment]
I'd like to validate the telemetry via an internal build before merging.  At the moment I'm blocked on not being able to clone build definitions in devdiv.

**Pull request merged**